concrete-csprng/src/counter/mod.rs

```diff
@@ -80,21 +84,27 @@ pub enum ShouldGenerateBatch {
 pub struct State {
     aes_ctr: AesCtr,
     byte_ctr: ByteCtr,
+    step_ctr: StepCtr,
```

Comment:
I definitely believe that this does not belong here. From my point of view, the state is an index to access a given byte. The step is not related to the state, but more with the way we increment it.